pkg/conditions/infrastructureready/spec.go

```diff
@@ -0,0 +1,38 @@
+package infrastructureready
+
+import (
+	"github.com/giantswarm/conditions/pkg/conditions"
+	corev1 "k8s.io/api/core/v1"
+	capi "sigs.k8s.io/cluster-api/api/v1alpha3"
+	capiexp "sigs.k8s.io/cluster-api/exp/api/v1alpha3"
+)
+
+type objectWithInfrastructureRef interface {
+	conditions.Object
+	GetInfrastructureRef() *corev1.ObjectReference
+	SetStatusInfrastructureReady(ready bool)
+}
+
+type clusterWrapper struct {
+	*capi.Cluster
+}
+
+func (c *clusterWrapper) GetInfrastructureRef() *corev1.ObjectReference {
+	return c.Spec.InfrastructureRef
+}
+
+func (c *clusterWrapper) SetStatusInfrastructureReady(value bool) {
+	c.Status.InfrastructureReady = value
```

Comment:
Setting this here as well, since it's basically deprecated by the condition itself, and upstream is setting it also together with the condition.